src/features/soft_navigations/aggregate/bel-node.js

```diff
@@ -14,9 +14,9 @@ export class BelNode {
   callbackDuration = 0
   nodeId = ++nodesSeen
 
-  constructor (agentIdentifier) {
-    if (!agentIdentifier) throw new Error('Interaction is missing core attributes')
-    this.agentIdentifier = agentIdentifier
+  constructor (agentRef) {
+    this.obfuscator = agentRef.runtime.obfuscator
+    this.info = agentRef.info
```

Comment:
These feel weird getting set in the base class, since it doesn't actually use anything from the agent ref. Could just set the relevant values in the outer classes -- Doesn't necessarily need to change, but just wanted to point that out

Reply:
Valid, but I'll save the duplication on the derived classes since they share using these assignments